tools/accuracy_checker/accuracy_checker/data_readers/data_reader.py

```diff
@@ -180,8 +179,163 @@ def read(self, data_id):
 class ScipyImageReader(BaseReader):
     __provider__ = 'scipy_imread'
 
+    @staticmethod
+    def _from_image(image, flatten=False, mode=None):
+        if mode is not None:
+            if mode != image.mode:
+                image = image.convert(mode)
+        elif image.mode == 'P':
+            image = image.convert('RGBA') if 'transparency' in image.info else image.convert('RGB')
+
+        if flatten:
+            image = image.convert('F')
+        elif image.mode == '1':
+            image = image.convert('L')
+
+        return np.array(image)
+
+    @staticmethod
+    def _process_2d(data, shape, mode, pal, high, low, cmax, cmin):
+        shape = (shape[1], shape[0])  # columns show up first
+        if mode == 'F':
+            data32 = data.astype(np.float32)
+            image = Image.frombytes(mode, shape, data32.tostring())
+
+            return image
+        if mode in [None, 'L', 'P']:
+            bytedata = ScipyImageReader._bytescale(data, high=high, low=low, cmin=cmin, cmax=cmax)
+            image = Image.frombytes('L', shape, bytedata.tostring())
+            if pal is not None:
+                image.putpalette(np.asarray(pal, dtype=np.uint8).tostring())
+                # Becomes a mode='P' automagically.
+            elif mode == 'P':  # default gray-scale
+                pal1 = np.arange(0, 256, 1, dtype=np.uint8)[:, np.newaxis]
+                pal2 = np.ones((3,), dtype=np.uint8)[np.newaxis, :]
+                pal = pal1 * pal2
+                image.putpalette(np.asarray(pal, dtype=np.uint8).tostring())
+
+            return image
+        if mode == '1':  # high input gives threshold for 1
+            bytedata = (data > high)
+            image = Image.frombytes('1', shape, bytedata.tostring())
+
+            return image
+
+        cmin = cmin or np.amin(np.ravel(data))
+        cmax = cmax or np.amax(np.ravel(data))
+        data = (data * 1.0 - cmin) * (high - low) / (cmax - cmin) + low
+        if mode == 'I':
+            data32 = data.astype(np.uint32)
+            image = Image.frombytes(mode, shape, data32.tostring())
+        else:
+            raise ValueError("Mode is unknown or incompatible with input array shape.")
+
+        return image
+
+    @staticmethod
+    def _process_3d(data, shape, mode, channel_axis, high, low, cmin, cmax):
```

Comment:
I don't think it's possible to reach this function, since it's only reachable through `imresize` and you only call `imresize` to do a 2-dimensional resize.

Reply:
Under 3d here meant image which has height, width, channels. I am not sure that it is unreachable, Pascal VOC annotation is 3-channels images before encoding, so if `resize_segmentation_mask` calls before encoding, it will process 3d image. I can not guarantee that it is not used in such way.

Reply:
I see, I was just confused about what 3D meant.